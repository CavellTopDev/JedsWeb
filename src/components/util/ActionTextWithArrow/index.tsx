import { SubtleButtonLink } from "@heydovetail/website/components/forms/SubtleButtonLink";
import IconArrowRightMini from "@heydovetail/website/components/icons/feather/IconArrowRightMini";
import { Flow } from "@heydovetail/website/components/layout/Flow";
import { Item } from "@heydovetail/website/components/layout/Item";
import { LocationDescriptor } from "@heydovetail/website/routing/locations";
import * as React from "react";

interface Props {
  location?: LocationDescriptor;
  text?: string;
}

export class ActionTextWithArrow extends React.PureComponent<Props> {
  public render() {
    const { location, text = "Learn more" } = this.props;
    const content = (
      <Flow styled={{ gap: 4 }}>
        <Item>{text}</Item>
        <Item>
          <IconArrowRightMini />
        </Item>
      </Flow>
    );

    if (location !== undefined) {
      return <SubtleButtonLink location={location}>{content}</SubtleButtonLink>;
    } else return content;
  }
}
