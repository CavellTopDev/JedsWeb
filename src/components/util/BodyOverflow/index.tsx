import React from "react";

interface Props {
  children?: React.ReactNode;
}

export class BodyOverflow extends React.Component<Props> {
  public componentDidMount() {
    window.document.body.style.overflow = "hidden";
  }

  public componentWillUnmount() {
    window.document.body.style.overflow = "auto";
  }

  public render() {
    return this.props.children !== undefined ? this.props.children : null;
  }
}
