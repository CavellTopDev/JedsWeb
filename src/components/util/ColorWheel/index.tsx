import React from "react";

interface Props {
  colors: string[];
}

interface State {
  color: number;
}

export class ColorWheel extends React.PureComponent<Props> {
  public state: State = {
    color: 0
  };

  private id = 0;

  public componentDidMount() {
    this.id = window.setInterval(this.timer, 2000);
  }

  public componentWillUnMount() {
    window.clearInterval(this.id);
  }

  private readonly timer = () => {
    this.setState({ color: this.state.color < this.props.colors.length - 1 ? this.state.color + 1 : 0 });
  };

  public render() {
    return (
      <span style={{ color: this.props.colors[this.state.color], transition: "color 500ms ease-in-out" }}>
        {this.props.children}
      </span>
    );
  }
}
