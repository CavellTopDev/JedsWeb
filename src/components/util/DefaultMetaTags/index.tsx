import * as thumbnail from "!file-loader!./thumbnail.png";
import * as React from "react";
import { Helmet } from "react-helmet";

// These are the default meta tags for every page.
// If a page uses Helmet to specify something more specific,
// it will override the meta tag here.

export class DefaultMetaTags extends React.PureComponent {
  public render() {
    return (
      <Helmet>
        <title>Dovetail — Customer feedback &amp; research platform for teams</title>
        <meta
          name="description"
          content="Dovetail combines customer feedback and user research in one place, making it easy to see patterns, discover insights, and decide what to do next. Our customers include Atlassian, Deloitte, Engima, HotelsCombined, Westpac, and more."
        />
        <meta
          name="keywords"
          content="customer feedback, data, data analysis, design software, design tool, dovetail, feedback, feedback management, product design, qualitative data analysis, research, research analysis, research repository, saas software, sydney australia, user research, ux design"
        />
        <meta name="twitter:card" content="summary" />
        <meta name="twitter:site" content="@hidovetail" />
        <meta property="og:title" content="A customer feedback &amp; research platform for teams — Dovetail" />
        <meta property="og:type" content="product" />
        <meta property="og:image" content={process.env.GATSBY_BASE_URL + thumbnail} />
        <meta
          property="og:description"
          content="Dovetail combines customer feedback and user research in one place, making it easy to see patterns, discover insights, and decide what to do next. Our customers include Atlassian, Deloitte, Engima, HotelsCombined, Westpac, and more."
        />
      </Helmet>
    );
  }
}
