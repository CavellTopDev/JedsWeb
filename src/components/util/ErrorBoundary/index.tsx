import { Container } from "@heydovetail/website/components/layout/Container";
import { WIDTH } from "@heydovetail/website/constants";
import * as React from "react";

interface State {
  error: boolean;
}

interface Props {
  children: React.ReactNode;
}

export class ErrorBoundary extends React.Component<Props, State> {
  public state: Readonly<State> = {
    error: false
  };

  public componentDidCatch(error: Error, errorInfo: { componentStack: string }) {
    this.setState({ error: true });
    // tslint:disable-next-line:no-typeof-undefined
    if (typeof Raven !== "undefined") {
      Raven.captureException(error, { extra: errorInfo });
    }
  }

  public render() {
    if (this.state.error) {
      return (
        <Container maxWidth={WIDTH} padding={{ x: 24, y: 128 }}>
          <p>
            Something has gone wrong. Please try each of the steps below to see if your problem is resolved, and if youâ€™re still
            having issues, contact us at <b>hello@dovetailapp.com</b>.
          </p>
          <ul>
            <li>
              <p>Refresh the current page</p>
            </li>
            <li>
              <p>Go back to the previous page</p>
            </li>
            <li>
              <p>Try another browser or device</p>
            </li>
          </ul>
        </Container>
      );
    } else {
      return this.props.children;
    }
  }
}
