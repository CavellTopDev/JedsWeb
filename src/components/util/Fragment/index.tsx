import * as React from "react";

export class Fragment extends React.Component {
  public render() {
    return this.props.children;
  }
}
