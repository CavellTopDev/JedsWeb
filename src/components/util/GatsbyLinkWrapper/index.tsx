import { Link } from "gatsby";
import React from "react";
import { LinkProps } from "react-router-dom";

export class GatsbyLinkWrapper extends React.PureComponent<LinkProps> {
  public render() {
    const { to } = this.props;

    let url = "";

    if (to.pathname !== undefined) {
      url = to.pathname;
    }

    if (to.search !== undefined) {
      url = `${url}?${to.search}`;
    }

    if (to.hash !== undefined) {
      url = `${url}#${to.hash}`;
    }

    return <Link to={url}>{this.props.children}</Link>;
  }
}
