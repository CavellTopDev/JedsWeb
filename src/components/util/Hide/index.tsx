import * as React from "react";
import * as MediaQuery from "react-responsive";
import { BREAKPOINT_PHABLET, BREAKPOINT_PHONE, BREAKPOINT_TABLET } from "../../../constants";

interface Props {
  children: React.ReactNode;
  device: "phone" | "phablet" | "tablet";
}

export class Hide extends React.PureComponent<Props> {
  public render() {
    return <MediaQuery query={`(min-width: ${this.width()}px)`}>{this.props.children}</MediaQuery>;
  }

  private width(): number {
    switch (this.props.device) {
      case "phone":
        return BREAKPOINT_PHONE;
      case "phablet":
        return BREAKPOINT_PHABLET;
      case "tablet":
        return BREAKPOINT_TABLET;
    }
  }
}
