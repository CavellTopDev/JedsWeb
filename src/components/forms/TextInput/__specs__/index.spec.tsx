import { assertReactSnapshot } from "dovetail/util/test";
import "jest";
import * as React from "react";
import { TextInput } from "../";

describe(TextInput.name, () => {
  describe("snapshots", () => {
    it("empty", () => {
      assertReactSnapshot(<TextInput />);
    });

    it("placeholder", () => {
      assertReactSnapshot(<TextInput placeholder="First name" />);
    });

    it("value", () => {
      assertReactSnapshot(<TextInput value="Bradley Ayers" />);
    });

    it("error", () => {
      assertReactSnapshot(<TextInput error="Invalid value" />);
    });

    it("on change", () => {
      assertReactSnapshot(<TextInput onChange={() => {}} />);
    });
  });
});
