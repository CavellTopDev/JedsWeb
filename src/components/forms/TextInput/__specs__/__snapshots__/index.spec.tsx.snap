// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshots empty 1`] = `
<input
  autoFocus={undefined}
  className="f259djl"
  defaultValue={undefined}
  maxLength={undefined}
  onBlur={undefined}
  onChange={undefined}
  onKeyDown={undefined}
  placeholder={undefined}
  value={undefined}
/>
`;

exports[`TextInput snapshots error 1`] = `
<span
  className="f19wd7ts"
>
  <input
    autoFocus={undefined}
    className="f259djl"
    defaultValue={undefined}
    maxLength={undefined}
    onBlur={undefined}
    onChange={undefined}
    onKeyDown={undefined}
    placeholder={undefined}
    value={undefined}
  />
  <span
    className="fwc669b"
  >
    Invalid value
  </span>
</span>
`;

exports[`TextInput snapshots on change 1`] = `
<input
  autoFocus={undefined}
  className="f259djl"
  defaultValue={undefined}
  maxLength={undefined}
  onBlur={undefined}
  onChange={[Function]}
  onKeyDown={undefined}
  placeholder={undefined}
  value={undefined}
/>
`;

exports[`TextInput snapshots placeholder 1`] = `
<input
  autoFocus={undefined}
  className="f259djl"
  defaultValue={undefined}
  maxLength={undefined}
  onBlur={undefined}
  onChange={undefined}
  onKeyDown={undefined}
  placeholder="First name"
  value={undefined}
/>
`;

exports[`TextInput snapshots value 1`] = `
<input
  autoFocus={undefined}
  className="f259djl"
  defaultValue={undefined}
  maxLength={undefined}
  onBlur={undefined}
  onChange={undefined}
  onKeyDown={undefined}
  placeholder={undefined}
  value="Bradley Ayers"
/>
`;
